Bash/InstallHuntress-macOS-bash.sh

```diff
@@ -61,22 +61,172 @@ install_system_extension=false
 ## Do not modify anything below this line
 ##############################################################################
 
-scriptVersion="August 15, 2024"
+scriptVersion="November 22, 2024"
 
-dd=$(date "+%Y%m%d-%H%M%S")
+dd=$(date "+%Y-%m-%d %H:%M:%S")
 log_file="/tmp/HuntressInstaller.log"
 install_script="/tmp/HuntressMacInstall.sh"
 invalid_key="Invalid account secret key"
 pattern="[a-f0-9]{32}"
-version="1.0"
+version="1.1"
 
-## Using logger function to provide helpful logs within RMM tools in addition to log file
+# print help message
+usage() {
+    cat <<EOF
+Usage: $0 [options...] --account_key=<account_key> --organization_key=<organization_key>
+
+-a, --account_key      <account_key>        The account key to use for this agent install
+-o, --organization_key <organization_key>   The org key to use for this agent install
+-t, --tags             <tags>               A comma-separated list of agent tags
+-i, --install_system_extension              If passed, automatically install the system extension
+-r, --reregister                            Attempt a clean slate install of Huntress (must also pass account and org keys)
+-h, --help                                  Print this message
+
+EOF
+}
+
+# Using logger function to provide helpful logs within RMM tools in addition to log file
 logger() {
     echo "$dd -- $*";
     echo "$dd -- $*" >> $log_file;
 }
 
-logger "Huntress install script last updated $scriptVersion"
+# This is for machines where the Huntress install is corrupted, damaged, or missing assets. 
+# This will try to fully remove Huntress before attempting a reinstall
+reregister() {
+    logger "Starting uninstall, please be aware this can take up to a minute"
+
+    countProcesses=$(ps aux | grep "HuntressAgent" -c)
+    if [ $countProcesses -le 1 ]; then
+        logger "Huntress process not running, System Extension may hang for a minute on uninstall"
```

Comment:
Is there a reason this message references the system extension specifically? Whether the Go agent (`HuntressAgent`) is running shouldn't have any effect on the removal of the extension. We'd need to check whether the Swift daemon (aka plain old `Huntress`) is running, since that's what's responsible for the extension.

Reply:
It was just something I noticed from testing. If I killed the Huntress processes before trying to remove the System Extension then the SE removal would pause for a period of time. I can adjust this to look for only plain Huntress instead.